Sources/TuistGraph/Models/Target.swift

```diff
@@ -193,93 +214,57 @@ public struct Target: Equatable, Hashable, Comparable, Codable {
-            return false
-        }
+        let isWatchOS = isExclusiveTo(.watchOS)
+        let isApp = (product == .watch2App || product == .app)
+        return isWatchOS && isApp
     }
 
     /// Determines if the target is an embeddable xpc service
     /// i.e. a product that can be bundled with a host macOS application
     public func isEmbeddableXPCService() -> Bool {
-        switch (platform, product) {
-        case (.macOS, .xpc):
-            return true
-        default:
-            return false
-        }
+        product == .xpc
     }
 
     /// Determines if the target is an embeddable system extension
     /// i.e. a product that can be bundled with a host macOS application
     public func isEmbeddableSystemExtension() -> Bool {
-        switch (platform, product) {
-        case (.macOS, .systemExtension):
-            return true
-        default:
-            return false
-        }
+        product == .systemExtension
     }
 
     /// Determines if the target is able to embed a watch application
     /// i.e. a product that can be bundled with a watchOS application
     public func canEmbedWatchApplications() -> Bool {
-        switch (platform, product) {
-        case (.iOS, .app):
-            return true
-        default:
-            return false
-        }
-    }
-
-    /// Determines if the target is able to embed an xpc service
-    /// i.e. a product that can be bundled with a macOS application
-    public func canEmbedXPCServices() -> Bool {
-        switch (platform, product) {
-        case (.macOS, .app):
-            return true
-        default:
-            return false
-        }
+        isExclusiveTo(.iOS) && product == .app
     }
 
     /// Determines if the target is able to embed an system extension
     /// i.e. a product that can be bundled with a macOS application
     public func canEmbedSystemExtensions() -> Bool {
-        switch (platform, product) {
-        case (.macOS, .app):
-            return true
-        default:
-            return false
-        }
+        isExclusiveTo(.macOS) && product == .app
     }
 
     /// For iOS targets that support macOS (Catalyst), this value is used
     /// in the generated build files of the target dependency products to
     /// indicate the build system that the dependency should be compiled
     /// with Catalyst compatibility.
-    public var targetDependencyBuildFilesPlatformFilter: BuildFilePlatformFilter? {
-        switch deploymentTarget {
-        case let .iOS(_, devices, _) where devices.contains(.all):
-            return nil
-        case let .iOS(_, devices, _):
-            if devices.contains(.mac) {
-                return .catalyst
+    public var dependencyPlatformFilters: PlatformFilters {
+        // is iOS only and has the equivalent of `.all` devices from `ProjectDescription.DeploymentTarget`
+        if isExclusiveTo(.iOS), destinations != Set([.iPad, .iPhone, .macCatalyst, .appleVisionWithiPadDesign]) {
```

Comment:
This may help remove the redundant iOS platform filter

```suggestion
        if isExclusiveTo(.iOS), destinations != .iOS {
```